src/core/model_config_utils.cc

```diff
@@ -603,81 +603,82 @@ GetNormalizedModelConfig(
     RETURN_IF_ERROR(ReadTextProto(config_path, config));
   }
 
-  // For now, both 'backend' and 'platform' cannot be listed in the
-  // config. If backend is given then just leave platform empty and
-  // don't perform autofill or attempt to adjust default model
-  // filename.
-  if (!config->platform().empty() && !config->backend().empty()) {
-    return Status(
-        Status::Code::INVALID_ARG,
-        "cannot specify both 'backend' and 'platform' for model '" +
-            config->name() + "'");
-  }
+  // Autofill if requested...
+  if (autofill) {
+    const std::string model_name(BaseName(path));
+    std::unique_ptr<AutoFill> af;
+    RETURN_IF_ERROR(AutoFill::Create(
+        model_name, backend_config_map, std::string(path), *config, &af));
+    RETURN_IF_ERROR(af->Fix(config));
 
-  if (config->backend().empty()) {
-    // Autofill if requested...
-    if (autofill) {
-      const std::string model_name(BaseName(path));
-      std::unique_ptr<AutoFill> af;
-      RETURN_IF_ERROR(AutoFill::Create(
-          model_name, backend_config_map, std::string(path), *config, &af));
-      RETURN_IF_ERROR(af->Fix(config));
-
-      LOG_VERBOSE(1) << "autofilled config: " << config->DebugString();
-    }
+    LOG_VERBOSE(1) << "autofilled config: " << config->DebugString();
+  }
 
-    if (config->platform().empty()) {
-      return Status(
-          Status::Code::INVALID_ARG,
-          "must specify platform for model '" + config->name() + "'");
```

Comment:
@deadeyegoodwin I changed the logic here and removed this check. The idea is that current autofill will simply do nothing for backend API (except for supported platform), so autofill path doesn't need to condition on `config->backend().empty()`. Then platform may still be empty after autofill.